app/presenters/redirect_presenter.rb

```diff
@@ -1,5 +1,46 @@
-module RedirectPresenter
-  def self.present(base_path:, publishing_app:, public_updated_at:, redirects:)
+class RedirectPresenter
+  def initialize(base_path:, publishing_app:, public_updated_at:, redirects:, content_id: nil, locale: nil)
+    @base_path = base_path
+    @publishing_app = publishing_app
+    @public_updated_at = public_updated_at
+    @redirects = redirects
+    @content_id = content_id
+    @locale = locale
+  end
+
+  def self.from_edition(edition)
+    new(
+      base_path: edition.base_path,
+      publishing_app: edition.publishing_app,
+      public_updated_at: edition.unpublishing.created_at,
+      redirects: edition.unpublishing.redirects,
+      content_id: edition.document.content_id,
+      locale: edition.document.locale,
+    )
+  end
+
+  def for_content_store(payload_version)
+    present.merge(payload_version: payload_version)
+  end
+
+  def for_message_queue
+    present.merge(
+      content_id: content_id,
+      locale: locale,
```

Comment:
cool thanks - do you also know if they're bothered about things not having a base_path?